Add doc that covers how to add an app from Git (#2935)
pipe-cd/pipe: docs/content/en/docs-dev/user-guide/adding-an-application.md

---

An application is a collect of resources and configurations that are managed together.
It represents the service which you are going to deploy. With PipeCD, all application's manifests and its application configuration (`app.pipecd.yaml`) must be committed into a directory of a Git repository. That directory is called as application directory.

> Before deploying an application, the application must be registered from either the web UI or an application configuration file defined in your Git repository.

Reviewer: I think both "manually form" and "git form" are registering via "web console". So maybe we don't have to explicitly distinguish them here.

```suggestion
Before deploying an application, it must be registered via the web console.
```

> Registering application helps PipeCD know the basic information about that application, where the application configuration is placing, what `piped` should handle it as well as what cloud the application should be deployed to.

Reviewer: ```suggestion
Registering application helps PipeCD know where the application configuration is placed, which `piped` should handle it as well as which cloud the application should be deployed to.
```

> An application can be handled by one of the registered `piped`s. Currently, PipeCD is supporting the following kinds of application:

Reviewer: I guess we have to update the Japanese doc too.

Author: Yeah, let me update them later 👍

> An application can be handled by one of the registered `piped`s. Currently, PipeCD is supporting the following kinds of application:

Reviewer: ```suggestion
Each application can be handled by one and only one `piped`. Currently, PipeCD is supporting the following application kinds:
```

- Kubernetes application
- Terraform application
- CloudRun application
- Lambda application
- ECS application

> There are two ways to register an application:
> - From the application configuration in your Git repository (recommended)
> - From the web UI

Reviewer: I want to say that both of them are "From the web UI", so I guess we need a better way to express them.
Below is just my suggestion.
We can find a better one.

```suggestion
There are two ways to register an application:
- Scanning the unused application configuration files in Git to add (recommended)
- Manually configure application information
```

## From the application configuration in your Git repository (recommended)
In this way, you define all information in the application configuration defined in the Git repository and use it as a single source of truth.

It starts with creating an application configuration file as following and pushing it to the Git repository watched by a Piped with version v0.23.0 or higher.
The file name must be suffixed by `.pipecd.yaml` because Piped periodically checks for files with this suffix.

```yaml
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  name: foo
```

> Visit [here](/user-guide/configuration-reference/) for supported fields.

Reviewer 2: ```suggestion
Visit [here](/docs/user-guide/configuration-reference/) for supported fields.
```
nits

Author: nice catch!

- `kind` is the application kind. As explained before, supporting kinds of application are: `Kubernetes`, `Terrform`, `CloudRun`, `Lambda` and `ECS`.
- `spec` is the specific configuration for each application kind.

Please refer [pipecd/examples](/docs/user-guide/examples/) for the deployments being supported.

The [next section](/docs/user-guide/configuring-deployment/) guides you how to configure the deployment for each specific application kinds.

## Updating an application

> Regardless of which method you use to register an application, you need to update the application via the application configuration file (that's why registering from Git is recommended).

Reviewer: ```suggestion
Regardless of which method you used to register the application, the web console can only be used to disable/enable/delete the application, besides the adding operation. All updates on application information must be done via the application configuration file stored in Git.
```

> Refer to [the supported field](/user-guide/configuration-reference/) and make revisions where you need to update.

Reviewer: ```suggestion
```

```yaml
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  name: new-name

> ```

Reviewer: ```suggestion
Refer to [configuration reference](/user-guide/configuration-reference/) to see the full list of configurable fields.
```

- `kind` is the application kind. As explained before, supporting kinds of application are: `Kubernetes`, `Terrform`, `CloudRun`, `Lambda` and `ECS`.
- `spec` is the specific configuration for each application kind.

Please refer [pipecd/examples](/docs/user-guide/examples/) for the deployments being supported.

The [next section](/docs/user-guide/configuring-deployment/) guides you how to configure the deployment for each specific application kinds.

## Updating an application
Regardless of which method you used to register the application, the web console can only be used to disable/enable/delete the application, besides the adding operation. All updates on application information must be done via the application configuration file stored in Git.

```yaml
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  name: new-name
```

> Refer to [configuration reference](/user-guide/configuration-reference/) to see the full list of configurable fields.

Reviewer 2: ```suggestion
Refer to [configuration reference](/docs/user-guide/configuration-reference/) to see the full list of configurable fields.
```
nits
